clients/js/src/revisions/v1/additionalSigner.ts

```diff
@@ -0,0 +1,14 @@
+import { RuleV2 } from '../v2';
```

Comment:
Could you make that an `import type` as well since it's out of the v1 and a parent file will import both v1 and v2, creating a circular dependency. Same for all other v1 rules.

```suggestion
import type { RuleV2 } from '../v2';
```